hive-metastore/src/main/java/org/apache/iceberg/hive/HiveOperationsBase.java

```diff
@@ -181,4 +264,220 @@ default Table newHmsTable(String hmsTableOwner) {
 
     return newTable;
   }
+
+  @SuppressWarnings("checkstyle:CyclomaticComplexity")
+  default void commitWithLocking(
+      Configuration conf,
+      BaseMetadata base,
+      BaseMetadata metadata,
+      String baseMetadataLocation,
+      String newMetadataLocation,
+      FileIO io) {
+    boolean newTable = base == null;
+    boolean hiveEngineEnabled = hiveEngineEnabled(conf, metadata);
```

Comment:
what about keepHiveStats?

Reply:
Thanks for catching this, I have added it back.

Reply:
@szehon-ho , Pl ignore my las comments. 
HiVeStatsCheck has been moved with [obsoleteProps](https://github.com/apache/iceberg/pull/9852/files#diff-5ecfc223b311b523a12be7482b6235318fdd7535c371f899daf7e09eacddad77R450) .

```
 if (!conf.getBoolean(ConfigProperties.KEEP_HIVE_STATS, false)) {
      obsoleteProps.add(StatsSetupConst.COLUMN_STATS_ACCURATE);
    }
```